scalafix-core/shared/src/main/scala/scalafix/internal/rule/Disable.scala

```diff
@@ -46,6 +46,36 @@ object Disable {
       buf.toList
     }
   }
+
+  def matchDisabledSymbol(disabledSymbol: DisabledSymbol, symbol: Symbol)(
+      implicit index: SemanticdbIndex): Boolean =
+    disabledSymbol match {
+      case DisabledSymbol(Some(s), _, _, banHierarchy, _) =>
+        SymbolOps.isSameNormalized(symbol, s) || (
+          banHierarchy &&
+            index
+              .denotation(symbol)
+              .exists(_.overrides.exists(SymbolOps.isSameNormalized(symbol, _)))
+        )
+      case d @ DisabledSymbol(None, _, _, _, Some(_)) =>
+        d.compiledRegex.exists(_.findFirstIn(symbol.toString).isDefined)
+      case DisabledSymbol(None, _, _, _, None) =>
+        true // weird case, but it has some logic
```

Comment:
Should this be handled by the ConfDecoder? If so then we might as well `sys.error("impossible")`.

Reply:
Comments like `// weird case, but it has some logic` are not very helpful. It's better to explain **why** this is a weird case and **what** "some logic" means.